asteroid/data/vad_dataset.py

```diff
@@ -0,0 +1,48 @@
+import json
+import soundfile as sf
+import torch
+from torch.utils.data import Dataset, DataLoader
+import random
+
+
+class VADDataset(Dataset):
+    """Dataset class for Voice Activity Detection.
+
+    Args:
+        md_file_path (str): The path to the metadata file.
+    """
+
+    def __init__(self, md_file_path, segment=3):
+
+        self.md_filepath = md_file_path
+        with open(self.md_filepath) as json_file:
+            self.md = json.load(json_file)
+        self.segment = segment
+
+    def __len__(self):
+        return len(self.md)
+
+    def __getitem__(self, idx):
+        # Get the row in dataframe
+        row = self.md[idx]
+        # Get mixture path
+        self.source_path = row[f"mixture_path"]
+        length = len(sf.read(self.source_path)[0])
+        if self.segment is not None:
+            start = random.randint(0, length - int(self.segment * 8000))
+            stop = start + int(self.segment * 8000)
+        else:
+            start = 0
+            stop = None
+
+        s, sr = sf.read(self.source_path, start=start, stop=stop, dtype="float32")
+        # Convert sources to tensor
+        source = torch.from_numpy(s)
+        label = self.from_vad_to_label(length, row["VAD"], start, stop).unsqueeze(0)
+        return source, label
+
+    def from_vad_to_label(self, length, vad, begin, end):
```

Comment:
Maybe this would be better moved out of the class into a function as it doesn't use `self`.